Amplify/Categories/DataStore/Query/ModelKey.swift

```diff
@@ -32,24 +32,43 @@ import Foundation
 ///     post.content != nil
 /// })
 /// ```
-public protocol ModelKey: CodingKey, CaseIterable, QueryFieldOperation {}
+public protocol ModelKey: CodingKey, CaseIterable, QueryFieldOperation, DefaultLogger {
+    var modelName: String { get }
+}
+
+extension ModelKey {
+    public var modelName: String { "" }
+}
 
 extension CodingKey where Self: ModelKey {
 
+    var columnName: String {
+        guard let modelSchema: ModelSchema = ModelRegistry.modelSchema(from: modelName) else {
```

Comment:
for customers that don't upgrade and re-codegen, how does modelName get resolved?

Reply:
oh i see above `public var modelName: String { "" }`